pagopa/io-app
# chore: [IOBP-935] Standardize error handling for user payment cancellation

## Short description
This pull request include adding a new error type for user payment cancellation on network failure.

## List of changes proposed in this pull request

* Added a new error type `ValidationFaultPaymentGenericErrorAfterUserCancellationProblemJson` to handle errors after user cancellation.
* Updated `WalletPaymentFailure` to include the new error type. 
* Modified the reducer to handle the new error type in `paymentsDeleteTransactionAction.failure`.

## How to test
- In _Pagamenti_
- Press on _Paga un avviso_
- Insert payments data and go at the latest step (**DO NOT PRESS ON PAY**)
- Press "Go Back" action repeatedly until a cancellation alert appears.
- Dismiss payment
